### PR TITLE
item stats - Fix Sanfew Serum and Revitalisation bonus calculations

### LINKED ISSUE #11539: Sanfew prayer restoration formula is wrong
It appears to be using lvl\*0.25+9, but after testing at a few thresholds I've found it to be lvl\*0.3+4.

Holy wrench adds +0.02 to the lvl mult (so prayer/super restore use 0.27, and sanfew would use 0.32)

Not sure which plugins this affects.

### PR DESCRIPTION
**Pull request by TheStonedTurtle**
Fixes #11539 

Before:
image
image

After:
image
image

Result after drinking dose with and without the wrench:
image

image

### CODE REVIEW

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ItemStatEffectTest
+{
+	private static final ImmutableMap<Integer, Integer> REVITALISATION_TABLE;
+
+	static
+	{
+		REVITALISATION_TABLE = new ImmutableMap.Builder<Integer, Integer>()
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:57]: Add a comment with the source of this data
```suggestion
		// https://oldschool.runescape.wiki/w/Revitalisation_(%2B)#Maximum_stat_restoration
		REVITALISATION_TABLE = new ImmutableMap.Builder<Integer, Integer>()
```

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 2]
```diff
@@ -78,6 +86,50 @@
 			.put(47, 25)  	.put(97, 40)
 			.put(49, 25)  	.put(99, 40)
 			.build();
+
+		SUPER_RESTORE_TABLE = new ImmutableMap.Builder<Integer, Integer>()
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:90]:
```suggestion
		// https://oldschool.runescape.wiki/w/Super_restore#Maximum_restoration
		SUPER_RESTORE_TABLE = new ImmutableMap.Builder<Integer, Integer>()
```

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 3]
```diff
@@ -78,6 +86,50 @@
...
+			.put(23, 13)  	.put(59, 22)  	.put(95, 31)
+			.put(24, 14)  	.put(60, 23)  	.put(96, 32)
+			.put(27, 14)  	.put(63, 23)  	.put(99, 32)
+			.put(28, 15)  	.put(64, 24)
+			.put(31, 15)  	.put(67, 24)
+			.put(32, 16)  	.put(68, 25)
+			.put(35, 16)  	.put(71, 25)
+			.build();
+
+		SANFEW_TABLE = new ImmutableMap.Builder<Integer, Integer>()
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:111]:
```suggestion
		// https://oldschool.runescape.wiki/w/Sanfew_serum#Maximum_restoration
		SANFEW_TABLE = new ImmutableMap.Builder<Integer, Integer>()
```

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 4]
```diff
@@ -93,15 +145,59 @@ public void prepare()
...
+	@Test
+	public void testSanfewSerumWithWrench()
+	{
+		prepareInventory();
+
+		final Effect item = itemStats.get(ItemID.SANFEW_SERUM1);
+		matchWikiTable(SANFEW_TABLE, item, true);
+	}
+
+	private void matchWikiTable(final ImmutableMap<Integer, Integer> table, final Effect item, boolean hasWrench)
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:198]: Given that the wrench test code is exactly the same as the implementation, I don't see a reason to test it.

**Comment by TheStonedTurtle** [comment]
PR commits have been updated as requested

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 5]
```diff
@@ -79,6 +86,52 @@
 			.put(47, 25)  	.put(97, 40)
 			.put(49, 25)  	.put(99, 40)
 			.build();
+
+		// https://oldschool.runescape.wiki/w/Super_restore#Maximum_restoration
+		SUPER_RESTORE_TABLE = new ImmutableMap.Builder<Integer, Integer>()
+			.put(1, 8)    	.put(36, 17)  	.put(72, 26)
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:92]: Nit-pick, but tabs shouldn't be included after the initial indentation. All of the alignment of table entries should be done using spaces. For instance, this line would be changed as follows:
```suggestion
			.put(1, 8)      .put(36, 17)    .put(72, 26)
```

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 6]
```diff
@@ -94,13 +147,38 @@ public void prepare()
 			.thenReturn(0);
 	}
 
+	private void prepareInventory()
```
Comment (Nightfirecat) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:150]: This looks like it's left behind from the holy wrench testing that was removed.

Comment (TheStonedTurtle) [reply runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java]: One of these force pushes will be right i promise.

File: runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+			.put(26, 11)    .put(59, 21)    .put(93, 31)
+			.put(27, 12)    .put(60, 22)    .put(94, 32)
+			.put(29, 12)    .put(63, 22)    .put(96, 32)
+			.put(30, 13)    .put(64, 23)    .put(97, 33)
+			.put(33, 13)    .put(66, 23)    .put(99, 33)
+			.build();
+	}
+
+	@Mock
+	@Bind
```
Comment (Adam-) [inline runelite-client/src/test/java/net/runelite/client/plugins/itemstats/ItemStatEffectTest.java:133]: this is only required if you are using the guice testlib stuff, and you aren't?

**Pull request merged**